### ISSUE TITLE
ct_expand:term/1 expands maps non-deterministically

### ISSUE THREAD

**Issue by jesperes** [uwiger/parse_trans#48]
When `ct_expand/1` expands maps, it calls `ct_expand:abstract_map/1`, which looks like this:
```erlang
abstract_map(Map) ->
    [{map_field_assoc,0,abstract(K),abstract(V)}
     || {K,V} <- maps:to_list(Map)
    ].
```
This means that the order of the keys in the `ct_expand`:ed map literal will depend on the order of the keys as returned by `maps:to_list/1`, which might be in any order. As a result, the generated beam will have a different md5-sum each time it is compiled, which means that e.g. `code:module_status/1` will always consider the beam "modified", even if no changes have been made.

I propose that the output from `maps:to_list/1` is passed through `lists:sort/1` before inserting it into the abstract code block.

**Comment by uwiger** [comment]
I agree that it's better to sort the map elements, if nothing else because it makes reading pretty-printed debug_info nicer.

I also understand that `maps:to_list/1` can produce different results for technically equal maps, if they were produced differently.

Out of curiosity: Are you saying that the order is non-deterministic even if the map was produced the same way each time?

**Comment by jesperes** [comment]
I think so, but I need to double-check to be 100% sure.

**Comment by jesperes** [comment]
Yes. I can confirm that when I compile the Erlang source file, I often get different orders. Sometimes they are the same, and some times there are only small differences, but some times the entire list is completely different. I guess that just means that it "might work if you're lucky".

**Comment by uwiger** [comment]
Does the above PR solve your problem?
